src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailTaskHistory.jsx

```diff
@@ -1,41 +1,149 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState } from 'react';
 import { useParams } from 'react-router-dom';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 
-import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import {
+  Alert, Button, DataTable,
+} from '@edx/paragon';
 import { getEmailTaskHistory } from './api';
+import messages from './messages';
 
-export default function BulkEmailTaskHistory() {
+export function BulkEmailTaskHistory({ intl }) {
   const { courseId } = useParams();
 
-  const [emailTaskHistoryData, setEmailTaskHistoryData] = useState(); // eslint-disable-line no-unused-vars
+  const [emailTaskHistoryData, setEmailTaskHistoryData] = useState();
+  const [showHistoricalTaskContentTable, setShowHistoricalTaskContentTable] = useState(false);
+  const [errorRetrievingData, setErrorRetrievingData] = useState(false);
 
-  useEffect(() => {
-    async function fetchEmailTaskHistoryData() {
-      const data = await getEmailTaskHistory(courseId);
+  /**
+   * Async function that makes a REST API call to retrieve historical bulk email (Instructor) task data for display
+   * within this component.
+   */
+  async function fetchEmailTaskHistoryData() {
+    let data = null;
+    try {
+      data = await getEmailTaskHistory(courseId);
+    } catch (error) {
+      setErrorRetrievingData(true);
+    }
+
+    if (data) {
       const { tasks } = data;
       setEmailTaskHistoryData(tasks);
+      setShowHistoricalTaskContentTable(true);
     }
-    fetchEmailTaskHistoryData();
-  }, []);
+  }
+
+  /**
+   * Render function for the email task history table. If an error occurs while attempting to fetch data from
+   * edx-platform we will render this error instead of the table.
+   */
+  const renderError = () => (
```

Comment:
I think `renderError` and `renderEmpty` could be the same function. Just pass the error level and the message when rendering.